dotnet-grad/Controllers/UserController.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using dotnet_grad.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace dotnet_grad.Controllers
+{
+  [ApiController]
+  [Route("api/[controller]")]
+  public class UserController : ControllerBase
+  {
+    TestContext testContext;
+    public UserController(TestContext _testContext)
+    {
+      testContext = _testContext;
+    }
+
+    [HttpGet("GetAllUsers")]
+    public List<UserModel> getUsers()
+    {
+      return testContext.Users.ToList();
```

Comment:
I'd highly recommend doing all DB transactions asynchronously as it avoid blocking the thread while DB operations are in progress. IE `.ToListAsync`, and also, please NEVER forget your cancellation tokens, they are generally passed through from the controller method IE `GetUsers(CancellationToken token)`. This will greatly improve the performance of your application.